src/protocol/message.rs

```diff
@@ -252,6 +256,7 @@ impl Message {
             }
             Message::Close(None) => Ok(String::new()),
             Message::Close(Some(frame)) => Ok(frame.reason.into_owned()),
+            Message::Frame(frame) => frame.into_string().map_err(Error::from),
```

Comment:
I btw wonder if `?` would work here (and at line 273 as well).

Reply:
Done.